### ISSUE TITLE
NavigationService is not setting parameterValues on request

### ISSUE THREAD

**Issue by Qwin** [MvvmCross/MvvmCross#2374]
Hi everyone had an issue today where the bundle that I gave as a parameter wasn't setting my Init method. After looking through the source code I found out why:

-  protected virtual async Task<TResult> Navigate<TParameter, TResult> (MvxViewModelRequest request, IMvxViewModel<TParameter, TResult> viewModel, TParameter param, IMvxBundle presentationBundle = null, CancellationToken cancellationToken = default(CancellationToken))

-  protected virtual async Task<TResult> Navigate<TResult>(MvxViewModelRequest request, IMvxViewModelResult<TResult> viewModel, IMvxBundle presentationBundle = null, CancellationToken cancellationToken = default(CancellationToken))

-   protected virtual async Task Navigate<TParameter>(MvxViewModelRequest request, IMvxViewModel<TParameter> viewModel, TParameter param, IMvxBundle presentationBundle = null, CancellationToken cancellationToken = default(CancellationToken))

-  protected virtual async Task Navigate(MvxViewModelRequest request, IMvxViewModel viewModel, IMvxBundle presentationBundle = null, CancellationToken cancellationToken = default(CancellationToken))

sourcecode source: https://github.com/MvvmCross/MvvmCross/blob/5ed5158910463998fc4df60bff51199ff153b8d8/MvvmCross/Core/Core/Navigation/MvxNavigationService.cs

All the above methods are not setting the PresentationBundle on the request properties before calling ShowViewModel. The PresentationBundle parameter is ignored as of now in those methods mentioned above. (Yes I could have created a PR if I wasn't so busy) As of now I am using the old ShowViewModel within the ViewModel to get it working. 

Expected : when using Navigate method in NavigationService I expect MvxBundle to set the Init Parameters in the ViewModel.

**Comment by nmilcoff** [comment]
I've also seen this some days ago. 

I already have some stashed changes to fix this issue. If nobody does it first, I'll open a PR after #2359 is merged, just to make it easier to review and also to avoid conflicts.

**Comment by nmilcoff** [comment]
This is almost a duplicate of #2228.

**Comment by nmilcoff** [comment]
@Qwin I think you're referring to `ParameterValues` instead of `PresentationValues`, is that right? Because PresentationValues seems to be always set in the ViewModelRequest.

**Comment by Qwin** [comment]
@nmilcoff yes you are right, I dont know why I said PresentationValues.

**Comment by sisterray** [comment]
Hi,

I have noticed that if I have a custom view presenter derived from MvxAppCompatViewPresenter, then the overridden version of  

`public override void ChangePresentation(MvxPresentationHint hint)`

is not called when I call `ShowViewModel` and pass a presentation hint. This worked in v4.2.3

I am trying to slowly migrate to v5.4.2 in a managed fashion. My first step was to upgrade the packages and get my existing application to work as before. So this is an issue for me. 

I was just wondering whether this is the same issue?

**Comment by nmilcoff** [comment]
That should be a different issue, can you open a new one and provide us
with a repro case?

**Comment by sisterray** [comment]
Sorry just realised it used to call `ShowFragment` not `ChangePresentation`. All working as it should.

**Comment by nmilcoff** [comment]
As we're moving forward with the new MvxNavigationService, we are not providing support for mixing navigation systems (ShowViewModel / MvxNavigationService.Navigate).
